pretext/Functions/Functionscancallotherfunctions.ptx

```diff
@@ -30,64 +30,110 @@
     </input>
   </program>
   <p>Even though this is a pretty simple idea, in practice this example illustrates many very
-    important Python concepts, including local and global variables along with parameter passing.
-    Note that the body of <c>square</c> is not executed until it is called from inside the <c>
-    sum_of_squares</c> function for the first time on line 6.</p>
-  <p>Also notice that when <c>square</c> is called (at Step 8, for example), there are two groups of
+    important Python concepts, including local variables, global variables, parameter passing, and how a larger function can be <alert>decomposed</alert>
+    into smaller sub-functions.The function <c>sum_of_squares</c> is <alert>composed</alert>of the <c>square</c>function and some addition.</p>
+    <p>
+    Note that the body of <c>square</c> is not executed until it is called from inside the <c>sum_of_squares</c> function for the first time on line 6.</p>
+
+  <p>Also notice that when <c>square</c> is called (at Steps 8, 13, and 18), there are two groups of
     local variables, one for <c>square</c> and one for <c>sum_of_squares</c>. Each group of local
     variables is called a <term>stack frame</term>. The variables <c>x</c>, and <c>y</c> are local
     variables in both functions. These are completely different variables, even though they have the
     same name. Each function invocation creates a new frame, and variables are looked up in that
-    frame. Notice that at step 9, y has the value 25 is one frame and 2 in the other.</p>
-  <p>What happens when you to refer to variable y on line 3? Python looks up the value of y in the
+    frame. Notice that at step 11, y has the value 25 is one frame and 2 in the other.</p>
+  <p>What happens when you to refer to variable y on line 3 at this step? Python looks up the value of y in the
     stack frame for the <c>square</c> function. If it didn't find it there, it would go look in the
     global frame.</p>
-  <!-- <p>Let's use composition to build up a little more useful function. Recall from the
-  dictionaries
-    chapter that we had a two-step process for finding the letter that appears most frequently in a
-    text string:</p>
+  
+  
+  
+  <p>Let's use composition to build up another function: finding the area of <alert>any</alert> regular polygon</p>
```

Comment:
I like this example, I just latexified it and changed the Python code from camelcase to snake for consistency.